File: 167_Two_sum_II.cpp
ass Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
     
        vector<int> ret;
        int _i_prev = numbers[0];
        int _j_prev = numbers[0];
        
        for(int i=0; i< numbers.size()-1; i++) {

            if(numbers[i] == _i_prev) continue;
            _i_prev = numbers[i];
            
            
            for(int j=i+1; j< numbers.size(); j++){
                if( (numbers[i] + numbers[j]) == target) {
                    ret.push_back(i+1);
                    ret.push_back(j+1);
                    return ret;
                }
            }
        }
        return ret;
    }
};

File: py/167-two-sum-ii.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        i = 0
        j = len(numbers) - 1

        while(numbers[i] + numbers[j] != target):
            if numbers[i] + numbers[j] > target:
                j -=1
            else:
                i +=1
        return [i+1, j+1]


if __name__ == "__main__":
    assert Solution().twoSum([2,7,11,15],9) == [1,2]
    assert Solution().twoSum([2,7,11,15],1) == []
    assert Solution().twoSum([],1) == []    